projects/demo/src/modules/markup/icons/icons-group/icons-group.component.ts

```diff
@@ -13,21 +13,24 @@ import {DemoTuiIconsList} from '../icons.tokens';
     changeDetection,
 })
 export class IconsGroupComponent {
-    iconsValues: Record<string, DemoTuiIconsList> = {};
+    @ContentChild(IconsGroupDirective)
+    readonly iconGroup?: IconsGroupDirective;
+
+    iconsValues: Record<string, readonly string[]> = {};
+    matcher = TUI_DEFAULT_MATCHER;
     keys: string[] = [];
     search = ``;
 
     @Input()
     @tuiRequiredSetter()
-    set icons(icons: Record<string, DemoTuiIconsList>) {
+    set icons(icons: Record<string, readonly string[]>) {
         this.keys = Object.keys(icons);
```

Comment:
but how we can provide icons from parent component?

Reply:
I mean instead of an input setter for `icons` and two props `keys` and `iconsValues` we can have one input property `icons` and a `@tuiPure getKeys(icons)` public method.